src/relay/backend/contrib/cmsisnn/relay_to_tir.cc

```diff
@@ -548,7 +560,21 @@ class RelayToTIRVisitor : public MixedModeMutator {
   }
 
   void EmitAdd(const GlobalVar& global_var, const Expr& expr) {
-    auto* add_call = expr.as<CallNode>();
+    const CallNode* clip_call = nullptr;
+    const CallNode* add_call = nullptr;
+    const CallNode* final_call = expr.as<CallNode>();
+    const OpNode* final_op = final_call->op.as<OpNode>();
+    int32_t output_min = std::numeric_limits<int8_t>::min();
+    int32_t output_max = std::numeric_limits<int8_t>::max();
+    if (final_op->name == "clip") {
```

Comment:
Can we refactor the parsing of binop --> clip into a function ?

As for a suggestion :
Something like : ParseBinaryElementwiseOpClipPattern(const Expr& expr) --> 
struct BinaryElementwiseClipPattern {
    Call binary_operator;
    Optional<Call> clip_operator;
}

So we can avoid duplicating the same code at two places.

Reply:
Done. Thanks @manupa-arm. That was indeed needed.